src/main/java/org/ibp/api/java/impl/middleware/dataset/DatasetExcelExportServiceImpl.java

```diff
@@ -56,8 +61,13 @@ public File export(final int studyId, final int datasetId, final Set<Integer> in
 
 	@Override
 	public List<MeasurementVariable> getColumns(final int studyId, final int datasetId, final boolean includeSampleGenotypeValues) {
-
-		final List<MeasurementVariable> columns = this.studyDatasetService.getSubObservationSetVariables(studyId, datasetId);
+		final DatasetDTO dataSet = this.datasetService.getDataset(datasetId);
```

Comment:
Maybe you can pass the datasetDto directly in getColumns abstract method, so that this.datasetService.getDataset() will not be executed multiple times, which is an expensive operation.